lib/Utils/Logger.cpp

```diff
@@ -51,74 +47,166 @@ std::string toString(const SeverityLevel &Level) {
     }
-    return boost::make_shared<std::ofstream>(LogFile);
-  }(LogFile);
-  Sink->locked_backend()->add_stream(Stream);
-  Sink->set_filter(&logFilter);
-  Sink->set_formatter(&logFormatter);
-  Sink->locked_backend()->auto_flush(true);
-  boost::log::core::get()->add_sink(Sink);
-  boost::log::core::get()->add_global_attribute(
-      "LineCounter", boost::log::attributes::counter<int>{});
-  boost::log::core::get()->add_global_attribute(
-      "Timestamp", boost::log::attributes::local_clock{});
-  boost::log::core::get()->set_exception_handler(
-      boost::log::make_exception_handler<std::exception>(
-          LoggerExceptionHandler()));
-  boost::log::core::get()->add_global_attribute("Duration", attrs::timer());
-#endif
+    return llvm::nulls();
+  }
+  auto StreamVariantIt = PassedLevelsToStreamVariant.find(Level);
+  if (StreamVariantIt != PassedLevelsToStreamVariant.end()) {
+    return getLogStreamFromStreamVariant(StreamVariantIt->second);
+  }
+  return llvm::nulls();
+}
+
+llvm::raw_ostream &Logger::getLogStreamFromStreamVariant(
+    const std::variant<StdStream, std::string> &StreamVariant) {
+  if (std::holds_alternative<StdStream>(StreamVariant)) {
+    auto StdStreamKind = std::get<StdStream>(StreamVariant);
+    if (StdStreamKind == StdStream::STDOUT) {
+      return llvm::outs();
+    }
+    if (StdStreamKind == StdStream::STDERR) {
+      return llvm::errs();
+    }
+    return llvm::nulls();
+  }
+  return *LogfileStreams[std::get<std::string>(StreamVariant)];
+}
+
+bool Logger::logCategory(const llvm::StringRef &Category,
+                         const std::optional<SeverityLevel> &Level) {
+  auto CategoryLookupIt = CategoriesToStreamVariant.find(Category);
+  if (CategoryLookupIt == CategoriesToStreamVariant.end()) {
+    return false;
+  }
+  if (Level.has_value()) {
+    for (const auto &[LevelThreshold, Stream] : CategoryLookupIt->second) {
+      if (LevelThreshold <= Level.value()) {
+        return true;
+      }
+    }
+    return false;
+  }
+  return CategoryLookupIt->second.count(Level) > 0;
 }
 
-void LoggerExceptionHandler::operator()(const std::exception &Ex) const {
-  std::cerr << "std::exception: " << Ex.what() << '\n';
+void Logger::addLinePrefix(llvm::raw_ostream &OS,
+                           const std::optional<SeverityLevel> &Level,
+                           const std::optional<std::string> &Category) {
+  // const auto NowTime = std::chrono::steady_clock::now();
+  // const auto MillisecondsDuration =
+  //     chrono::duration_cast<chrono::milliseconds>(NowTime -
+  //     StartTime).count();
+  // OS << MillisecondsDuration;
+  if (Level.has_value()) {
+    OS << '[' << toString(Level.value()) << ']';
+  } // else {
+    // OS << ' ';
+    // }
+  if (Category.has_value()) {
+    OS << '[' << Category.value() << ']';
+  } // else {
+  //   OS << ' ';
+  // }
+  OS << ' ';
 }
 
+void initializeLogger(bool UseLogger, const std::string &LogFile) {}
```

Comment:
Remove this one?

Reply:
it actually needed an implementation, I have deprecated this function and we will remove it in the overnext release to allow our users to migrate.